### ISSUE TITLE
Question: MC Samples of GGN

### ISSUE THREAD

**Issue by perone** [f-dangel/backpack#21]
Hi, thanks a lot for open sourcing backpack, it's a **major contribution** for the community. I wasn't able to find a better place to ask questions, so here we go:

1) The MC samples of the GGN are taken from the model predictive distribution (y ~ p_{\theta}(x)) and not from the empirical data right ? Is the target of the loss ignored while sampling to compute the GGN then ?
2) What is the easiest way to get more MC samples ? Multiple `backward()` calls with `zero_grad()` calls ?

Thanks !!

**Comment by f-dangel** [comment]
Hi, thanks for your feedback. We're glad you like the package!

Regarding your questions:
1. We sample from the model's predictive distribution and not from the empirical data ([this](https://github.com/f-dangel/backpack/blob/development/backpack/core/derivatives/crossentropyloss.py#L30-L45) is the underlying code for cross entropy, the loss is ignored)
2. I coded up an experimental quick-shot for setting the number of samples in `DiagGGNMC` and `KFAC` on [this branch](https://github.com/f-dangel/backpack/tree/mc-samples). With that, you should be able to call (caution: the code is untested)
```python
mc_samples = 10 # defaults to 1
with backpack(DiagGGNMC(mc_samples)):
    loss.backward()
```
(same for `KFAC`)

**Comment by perone** [comment]
Thanks @f-dangel ! The code for sampling is very elegant.
